Source/Shared/BPUtils.m

```diff
@@ -121,6 +121,28 @@ + (NSError *)BPError:(const char *)function andLine:(int)line withFormat:(NSStri
                            userInfo:@{NSLocalizedDescriptionKey: msg}];
 }
 
++ (NSString *)findExecutablePath:(NSString *)execName {
+    NSString *argv0 = [[[NSProcessInfo processInfo] arguments] objectAtIndex:0];
+    NSString *execPath = [[argv0 stringByDeletingLastPathComponent] stringByAppendingPathComponent:execName];
+    if (![[NSFileManager defaultManager] isExecutableFileAtPath:execPath]) {
```

Comment:
How does `isExecutableAtPath` find a `.dylib`? Are you making the `.dylib` executable?